### ISSUE TITLE
error loading RCurl

### ISSUE THREAD

**Issue by dtenenba** [paws-r/paws#207]
In a fresh session, I can load the `RCurl` package just fine, but when `paws` loads it, I get an error.

Fresh session:

```
> library(RCurl)
Loading required package: bitops
>
```

From `paws`:

```
>  library(paws);s3 <- paws::s3();s3$list_buckets()
Error in dyn.load(file, DLLpath = DLLpath, ...) :
  unable to load shared object '/home/dtenenba/R/x86_64-pc-linux-gnu-library/3.6/curl/libs/curl.so':
  /app/easybuild/software/cURL/7.49.1-foss-2016b/lib/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by /home/dtenenba/R/x86_64-pc-linux-gnu-library/3.6/curl/libs/curl.so)
> traceback()
18: dyn.load(file, DLLpath = DLLpath, ...)
17: library.dynam(lib, package, package.lib)
16: loadNamespace(name)
15: getNamespace(ns)
14: asNamespace(ns)
13: getExportedValue(pkg, name)
12: curl::new_handle
11: handle(name)
10: handle_find(url)
9: handle_url(handle, url, ...)
8: httr::VERB(method, url = url, config = httr::add_headers(.headers = headers),
       body = body)
7: issue(request$http_request)
6: sender(request)
5: handler$fn(request)
4: run(request, send)
3: send(request)
2: send_request(request)
1: s3$list_buckets()
> sessionInfo()
R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 14.04.5 LTS

Matrix products: default
BLAS/LAPACK: /app/easybuild/software/OpenBLAS/0.2.18-GCC-5.4.0-2.26-LAPACK-3.6.1/lib/libopenblas_prescottp-r0.2.18.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] paws_0.1.4      RCurl_1.95-4.12 bitops_1.0-6

loaded via a namespace (and not attached):
[1] httr_1.4.1         compiler_3.6.1     R6_2.4.0           ini_0.3.1
[5] paws.storage_0.1.4 digest_0.6.20      paws.common_0.2.2
```

**Comment by davidkretch** [comment]
Sorry about that.  Are you able to run `curl::curl_fetch_memory("http://httpbin.org/cookies/set?foo=123&bar=ftw")`?

**Comment by dtenenba** [comment]
Yes.

```
> curl::curl_fetch_memory("http://httpbin.org/cookies/set?foo=123&bar=ftw")
$url
[1] "http://httpbin.org/cookies"

$status_code
[1] 200

$type
[1] "application/json"

$headers
  [1] 48 54 54 50 2f 31 2e 31 20 33 30 32 20 46 4f 55 4e 44 0d 0a 41 63 63 65 73
 [26] 73 2d 43 6f 6e 74 72 6f 6c 2d 41 6c 6c 6f 77 2d 43 72 65 64 65 6e 74 69 61
 [51] 6c 73 3a 20 74 72 75 65 0d 0a 41 63 63 65 73 73 2d 43 6f 6e 74 72 6f 6c 2d
 [76] 41 6c 6c 6f 77 2d 4f 72 69 67 69 6e 3a 20 2a 0d 0a 43 6f 6e 74 65 6e 74 2d
[101] 54 79 70 65 3a 20 74 65 78 74 2f 68 74 6d 6c 3b 20 63 68 61 72 73 65 74 3d
[126] 75 74 66 2d 38 0d 0a 44 61 74 65 3a 20 54 75 65 2c 20 31 35 20 4f 63 74 20
[151] 32 30 31 39 20 31 34 3a 33 33 3a 35 34 20 47 4d 54 0d 0a 4c 6f 63 61 74 69
[176] 6f 6e 3a 20 2f 63 6f 6f 6b 69 65 73 0d 0a 52 65 66 65 72 72 65 72 2d 50 6f
[201] 6c 69 63 79 3a 20 6e 6f 2d 72 65 66 65 72 72 65 72 2d 77 68 65 6e 2d 64 6f
[226] 77 6e 67 72 61 64 65 0d 0a 53 65 72 76 65 72 3a 20 6e 67 69 6e 78 0d 0a 53
[251] 65 74 2d 43 6f 6f 6b 69 65 3a 20 66 6f 6f 3d 31 32 33 3b 20 50 61 74 68 3d
[276] 2f 0d 0a 53 65 74 2d 43 6f 6f 6b 69 65 3a 20 62 61 72 3d 66 74 77 3b 20 50
[301] 61 74 68 3d 2f 0d 0a 58 2d 43 6f 6e 74 65 6e 74 2d 54 79 70 65 2d 4f 70 74
[326] 69 6f 6e 73 3a 20 6e 6f 73 6e 69 66 66 0d 0a 58 2d 46 72 61 6d 65 2d 4f 70
[351] 74 69 6f 6e 73 3a 20 44 45 4e 59 0d 0a 58 2d 58 53 53 2d 50 72 6f 74 65 63
[376] 74 69 6f 6e 3a 20 31 3b 20 6d 6f 64 65 3d 62 6c 6f 63 6b 0d 0a 43 6f 6e 74
[401] 65 6e 74 2d 4c 65 6e 67 74 68 3a 20 32 32 33 0d 0a 43 6f 6e 6e 65 63 74 69
[426] 6f 6e 3a 20 6b 65 65 70 2d 61 6c 69 76 65 0d 0a 0d 0a 48 54 54 50 2f 31 2e
[451] 31 20 32 30 30 20 4f 4b 0d 0a 41 63 63 65 73 73 2d 43 6f 6e 74 72 6f 6c 2d
[476] 41 6c 6c 6f 77 2d 43 72 65 64 65 6e 74 69 61 6c 73 3a 20 74 72 75 65 0d 0a
[501] 41 63 63 65 73 73 2d 43 6f 6e 74 72 6f 6c 2d 41 6c 6c 6f 77 2d 4f 72 69 67
[526] 69 6e 3a 20 2a 0d 0a 43 6f 6e 74 65 6e 74 2d 45 6e 63 6f 64 69 6e 67 3a 20
[551] 67 7a 69 70 0d 0a 43 6f 6e 74 65 6e 74 2d 54 79 70 65 3a 20 61 70 70 6c 69
[576] 63 61 74 69 6f 6e 2f 6a 73 6f 6e 0d 0a 44 61 74 65 3a 20 54 75 65 2c 20 31
[601] 35 20 4f 63 74 20 32 30 31 39 20 31 34 3a 33 33 3a 35 34 20 47 4d 54 0d 0a
[626] 52 65 66 65 72 72 65 72 2d 50 6f 6c 69 63 79 3a 20 6e 6f 2d 72 65 66 65 72
[651] 72 65 72 2d 77 68 65 6e 2d 64 6f 77 6e 67 72 61 64 65 0d 0a 53 65 72 76 65
[676] 72 3a 20 6e 67 69 6e 78 0d 0a 58 2d 43 6f 6e 74 65 6e 74 2d 54 79 70 65 2d
[701] 4f 70 74 69 6f 6e 73 3a 20 6e 6f 73 6e 69 66 66 0d 0a 58 2d 46 72 61 6d 65
[726] 2d 4f 70 74 69 6f 6e 73 3a 20 44 45 4e 59 0d 0a 58 2d 58 53 53 2d 50 72 6f
[751] 74 65 63 74 69 6f 6e 3a 20 31 3b 20 6d 6f 64 65 3d 62 6c 6f 63 6b 0d 0a 43
[776] 6f 6e 74 65 6e 74 2d 4c 65 6e 67 74 68 3a 20 36 38 0d 0a 43 6f 6e 6e 65 63
[801] 74 69 6f 6e 3a 20 6b 65 65 70 2d 61 6c 69 76 65 0d 0a 0d 0a

$modified
[1] NA

$times
     redirect    namelookup       connect   pretransfer starttransfer
     0.203288      0.000035      0.000038      0.000080      0.087079
        total
     0.290403

$content
 [1] 7b 0a 20 20 22 63 6f 6f 6b 69 65 73 22 3a 20 7b 0a 20 20 20 20 22 62 61 72
[26] 22 3a 20 22 66 74 77 22 2c 20 0a 20 20 20 20 22 66 6f 6f 22 3a 20 22 31 32
[51] 33 22 0a 20 20 7d 0a 7d 0a

>
```

**Comment by dtenenba** [comment]
Yes.

```
> curl::curl_fetch_memory("http://httpbin.org/cookies/set?foo=123&bar=ftw")
$url
[1] "http://httpbin.org/cookies"

$status_code
[1] 200

$type
[1] "application/json"

$headers
  [1] 48 54 54 50 2f 31 2e 31 20 33 30 32 20 46 4f 55 4e 44 0d 0a 41 63 63 65 73
 [26] 73 2d 43 6f 6e 74 72 6f 6c 2d 41 6c 6c 6f 77 2d 43 72 65 64 65 6e 74 69 61
 [51] 6c 73 3a 20 74 72 75 65 0d 0a 41 63 63 65 73 73 2d 43 6f 6e 74 72 6f 6c 2d
 [76] 41 6c 6c 6f 77 2d 4f 72 69 67 69 6e 3a 20 2a 0d 0a 43 6f 6e 74 65 6e 74 2d
[101] 54 79 70 65 3a 20 74 65 78 74 2f 68 74 6d 6c 3b 20 63 68 61 72 73 65 74 3d
[126] 75 74 66 2d 38 0d 0a 44 61 74 65 3a 20 54 75 65 2c 20 31 35 20 4f 63 74 20
[151] 32 30 31 39 20 31 34 3a 33 33 3a 35 34 20 47 4d 54 0d 0a 4c 6f 63 61 74 69
[176] 6f 6e 3a 20 2f 63 6f 6f 6b 69 65 73 0d 0a 52 65 66 65 72 72 65 72 2d 50 6f
[201] 6c 69 63 79 3a 20 6e 6f 2d 72 65 66 65 72 72 65 72 2d 77 68 65 6e 2d 64 6f
[226] 77 6e 67 72 61 64 65 0d 0a 53 65 72 76 65 72 3a 20 6e 67 69 6e 78 0d 0a 53
[251] 65 74 2d 43 6f 6f 6b 69 65 3a 20 66 6f 6f 3d 31 32 33 3b 20 50 61 74 68 3d
[276] 2f 0d 0a 53 65 74 2d 43 6f 6f 6b 69 65 3a 20 62 61 72 3d 66 74 77 3b 20 50
[301] 61 74 68 3d 2f 0d 0a 58 2d 43 6f 6e 74 65 6e 74 2d 54 79 70 65 2d 4f 70 74
[326] 69 6f 6e 73 3a 20 6e 6f 73 6e 69 66 66 0d 0a 58 2d 46 72 61 6d 65 2d 4f 70
[351] 74 69 6f 6e 73 3a 20 44 45 4e 59 0d 0a 58 2d 58 53 53 2d 50 72 6f 74 65 63
[376] 74 69 6f 6e 3a 20 31 3b 20 6d 6f 64 65 3d 62 6c 6f 63 6b 0d 0a 43 6f 6e 74
[401] 65 6e 74 2d 4c 65 6e 67 74 68 3a 20 32 32 33 0d 0a 43 6f 6e 6e 65 63 74 69
[426] 6f 6e 3a 20 6b 65 65 70 2d 61 6c 69 76 65 0d 0a 0d 0a 48 54 54 50 2f 31 2e
[451] 31 20 32 30 30 20 4f 4b 0d 0a 41 63 63 65 73 73 2d 43 6f 6e 74 72 6f 6c 2d
[476] 41 6c 6c 6f 77 2d 43 72 65 64 65 6e 74 69 61 6c 73 3a 20 74 72 75 65 0d 0a
[501] 41 63 63 65 73 73 2d 43 6f 6e 74 72 6f 6c 2d 41 6c 6c 6f 77 2d 4f 72 69 67
[526] 69 6e 3a 20 2a 0d 0a 43 6f 6e 74 65 6e 74 2d 45 6e 63 6f 64 69 6e 67 3a 20
[551] 67 7a 69 70 0d 0a 43 6f 6e 74 65 6e 74 2d 54 79 70 65 3a 20 61 70 70 6c 69
[576] 63 61 74 69 6f 6e 2f 6a 73 6f 6e 0d 0a 44 61 74 65 3a 20 54 75 65 2c 20 31
[601] 35 20 4f 63 74 20 32 30 31 39 20 31 34 3a 33 33 3a 35 34 20 47 4d 54 0d 0a
[626] 52 65 66 65 72 72 65 72 2d 50 6f 6c 69 63 79 3a 20 6e 6f 2d 72 65 66 65 72
[651] 72 65 72 2d 77 68 65 6e 2d 64 6f 77 6e 67 72 61 64 65 0d 0a 53 65 72 76 65
[676] 72 3a 20 6e 67 69 6e 78 0d 0a 58 2d 43 6f 6e 74 65 6e 74 2d 54 79 70 65 2d
[701] 4f 70 74 69 6f 6e 73 3a 20 6e 6f 73 6e 69 66 66 0d 0a 58 2d 46 72 61 6d 65
[726] 2d 4f 70 74 69 6f 6e 73 3a 20 44 45 4e 59 0d 0a 58 2d 58 53 53 2d 50 72 6f
[751] 74 65 63 74 69 6f 6e 3a 20 31 3b 20 6d 6f 64 65 3d 62 6c 6f 63 6b 0d 0a 43
[776] 6f 6e 74 65 6e 74 2d 4c 65 6e 67 74 68 3a 20 36 38 0d 0a 43 6f 6e 6e 65 63
[801] 74 69 6f 6e 3a 20 6b 65 65 70 2d 61 6c 69 76 65 0d 0a 0d 0a

$modified
[1] NA

$times
     redirect    namelookup       connect   pretransfer starttransfer
     0.203288      0.000035      0.000038      0.000080      0.087079
        total
     0.290403

$content
 [1] 7b 0a 20 20 22 63 6f 6f 6b 69 65 73 22 3a 20 7b 0a 20 20 20 20 22 62 61 72
[26] 22 3a 20 22 66 74 77 22 2c 20 0a 20 20 20 20 22 66 6f 6f 22 3a 20 22 31 32
[51] 33 22 0a 20 20 7d 0a 7d 0a

>
```

**Comment by davidkretch** [comment]
Hi, sorry for the delay.  We are having a hard time with this bug.  Are you able to run `httr::GET(url="https://httpbin.org/get")`?

**Comment by dtenenba** [comment]
No:

```
> httr::GET(url="https://httpbin.org/get")
Error in dyn.load(file, DLLpath = DLLpath, ...) :
  unable to load shared object '/home/dtenenba/R/x86_64-pc-linux-gnu-library/3.6/curl/libs/curl.so':
  /app/easybuild/software/cURL/7.49.1-foss-2016b/lib/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by /home/dtenenba/R/x86_64-pc-linux-gnu-library/3.6/curl/libs/curl.so)
> traceback()
11: dyn.load(file, DLLpath = DLLpath, ...)
10: library.dynam(lib, package, package.lib)
9: loadNamespace(name)
8: getNamespace(ns)
7: asNamespace(ns)
6: getExportedValue(pkg, name)
5: curl::new_handle
4: handle(name)
3: handle_find(url)
2: handle_url(handle, url, ...)
1: httr::GET(url = "https://httpbin.org/get")
> sessionInfo()
R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 14.04.5 LTS

Matrix products: default
BLAS/LAPACK: /app/easybuild/software/OpenBLAS/0.2.18-GCC-5.4.0-2.26-LAPACK-3.6.1/lib/libopenblas_prescottp-r0.2.18.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

loaded via a namespace (and not attached):
[1] httr_1.4.1     compiler_3.6.1 R6_2.4.0
```

It may be something specific to my setup. In any case it's not a `paws` issue so you may not want to spend too much time on it.

**Comment by davidkretch** [comment]
Sorry.  In any case, we need better documentation about OS-level dependencies.  Please let us know if you figure out what the issue is.

I've tried it on Ubuntu 14.04 with the following packages.  I tried each of the three available `libcurl` packages, and had to reinstall the `curl` R package after.  The version of cURL available in the public repo is 7.35.0; I am not sure if it is critically different than 7.49.1.

| package              | version                 |
|----------------------|-------------------------|
| libcurl4-openssl-dev | 7.35.0-1ubuntu2.20      |
| libcurl4-nss-dev     | 7.35.0-1ubuntu2.20      |
| libcurl4-gnutls-dev  | 7.35.0-1ubuntu2.20      |
| libssl-dev           | 1.0.1f-1ubuntu2.27      |
| libxml2-dev          | 2.9.1+dfsg1-3ubuntu4.13 |

**Comment by davidkretch** [comment]
Added lists of required OS-level packages for Linux to the readme.
